rclc/src/rclc/node.c

```diff
@@ -79,6 +79,24 @@ rclc_node_init_with_options(
     node_ops);
   if (rc != RCL_RET_OK) {
     PRINT_RCLC_WARN(rclc_node_init_with_options, rcl_node_init);
+    return rc;
   }
+
+  // The initialization for the rosout publisher
+  if (rcl_logging_rosout_enabled() && node_ops->enable_rosout) {
+    rc = rcl_logging_rosout_init_publisher_for_node(node);
+    if (rc != RCL_RET_OK) {
+      PRINT_RCLC_WARN(
+        rclc_node_init_with_options,
+        rcl_logging_rosout_init_publisher_for_node);
+      if (rcl_logging_rosout_fini_publisher_for_node(node) != RCL_RET_OK) {
+        PRINT_RCLC_WARN(
+          rclc_node_init_with_options,
+          rcl_logging_rosout_fini_publisher_for_node);
+      }
+      return rc;
+    }
+  }
+
```

Comment:
As there is no `rclc_node_fini`method, users currenlty clean up with `rcl_node_fini` like in  this code example [example_executor.c#L172](https://github.com/ros2/rclc/blob/d263be2057e39f1f702f014548c60f06b5a51333/rclc_examples/src/example_executor.c#L172C1-L172C33).

I do not see  `rcl_logging_rosout_fini_publisher_for_node` called in [`rcl_node_fini`](https://github.com/ros2/rcl/blob/rolling/rcl/src/rcl/node.c).

@fujitatomoya Would it make sense to cleanup this rosout publisher in `rcl_node_fini`? Or would it make more sense to create a `rclc_node_fini` method in rclc library?